eclass/zig-toolchain.eclass

```diff
@@ -0,0 +1,382 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: zig-toolchain.eclass
```

Comment:
To be honest, I think this is a bad name. Given precedence, it suggests that this is an eclass for *building Zig toolchain*.

Reply:
Why not `zig.eclass`? `rust-toolchain.eclass` is primarily used for internals by `dev-lang/rust`.

Reply:
I have renamed "zig-toolchain" to "zig-utils", is it good enough?

> Why not zig.eclass

I wanted to emphasize that "zig-build" eclass supports mainly build.zig and build.zig.zon based projects, with official package manager, not https://github.com/nektro/zigmod and similar

f1907ac58911...35373ea63ba5

Reply:
I don't think that's a useful distinction to make for Gentoo.

Reply:
Ok, will rename so

Reply:
d5fd6142e3fb...966f169f3fb2